docker/docker_client.go

```diff
@@ -165,8 +165,13 @@ func dockerCertDir(sys *types.SystemContext, hostPort string) (string, error) {
 	var (
 		hostCertDir     string
 		fullCertDirPath string
+		homeCertDir     string
 	)
-	for _, systemPerHostCertDirPath := range systemPerHostCertDirPaths {
+	home := os.Getenv("HOME")
+	if home != "" {
+		homeCertDir = filepath.Join(home, ".config", "containers", ".certs.d")
+	}
+	for _, systemPerHostCertDirPath := range append([]string{homeCertDir}, systemPerHostCertDirPaths...) {
```

Comment:
- I don’t think this works correctly for `homeCertDir == ""`
- `RootForImplicitAbsolutePaths` should not be used for the home directory.
- We use `docker/docker/pkg/homedir` when looking up credentials, it would be nice to be consistent.

Reply:
Fixed 1 and 3, not sure we should hack around 2.

Reply:
Yes, 2. still needs handling correctly. (See `legacyFormat` in `GetAuthentication` for one way to structure the code without repeating the consuming loop.)

Reply:
I think I have it the way you want it now.